### ISSUE TITLE
About Dataset conformity 

### ISSUE THREAD

**Issue by IGNCNIG** [INSPIRE-MIF/helpdesk-validator#2]
Hello,

we have problems with the declaration of conformity in our metadata  when previously it was not.

We inform about the name of the Regulation "REGLAMENTO (UE) No 1089/2010 DE LA COMISIÓN de 23 de noviembre de 2010 por el que se aplica la Directiva 2007/2/CE del Parlamento Europeo y del Consejo en lo que se refiere a la interoperabilidad de los conjuntos y los servicios de datos espaciales" , the date "2010-12-08" and the type "publication".

This an example of metadata that include this information in the section of Conformity:
http://www.ign.es/csw-inspire/srv/spa/csw?SERVICE=CSW&VERSION=2.0.2&REQUEST=GetRecordById&outputSchema=http://www.isotc211.org/2005/gmd&ElementSetName=full&ID=spaignLLM2013

The messages  of the mistake in the validator says "One of the citation elements shall contain title and date of the [Regulation 1089/2010]"  but these information are completed.

Why is not confomed this metadata file? What is the problem? 

Thanks you

Best Regards

Alejandra

**Comment by inakidiazdecerio** [comment]
Dear @IGNCNIG,

Thank you for reporting the issue, we have been analysing it. There have been some [modifications ](https://github.com/inspire-eu-validation/metadata/pull/137/files/9b496bda470952e1a801d590b91cb09343e3acf2)in the Dataset Conformity test, that's the reason the same metadata file is giving a new error message. 

The title text in the CI_Citation should be the title in the relevant MD language. The correct titles in the different languages are available in the language-specific files at http://inspire.ec.europa.eu/schemas/common/1.0/enums/

In the case of the title of [Spain](http://inspire.ec.europa.eu/schemas/common/1.0/enums/enum_spa.xsd) the title text is apparently wrong in the enumeration: 

`REGLAMENTO (UE) N o 1089/2010 DE LA COMISIÓN de 23 de noviembre de 2010 por el que se aplica la Directiva 2007/2/CE del Parlamento Europeo y del Consejo en lo que se refiere a la interoperabilidad de los conjuntos y los servicios de datos espaciales`

Anyway, it was suggested to allow different capitalisation of words and also additional spaces. At this moment, only additional spaces are allowed at the beginning and at the end of the title.

Summing up, we will propose following actions and come back to you with the solution:

- Modify the enumeration of Spain
- Find a solution for the interleaved whitespaces in the string

**Comment by michellutz** [comment]
@aquaglia The typo also needs to be fixed in the schema http://inspire.ec.europa.eu/schemas/common/1.0/enums/

**Comment by michellutz** [comment]
@aquaglia  To be precise, a space needs to be removed between "N" and "o" in the enumeration value "REGLAMENTO (UE) N o 1089/2010 ..." in http://inspire.ec.europa.eu/schemas/common/1.0/enums/enum_spa.xsd

**Comment by michellutz** [comment]
We would like to discuss in the sub-group meeting tomorrow, how strict/flexible the comparison of free text elements like this one should be. The current proposal is to ignore whitespaces in the comparison (also in the middle of the text), but not to ignore punctuation (e.g. additional commas). Another approach could be to be strict and require the text to be provided exactly as required, allowing only for different capitalisation.

**Comment by aquaglia** [comment]
I think we should take into consideration two additional aspects:
- different languages and alphabets might give whitespace and capitalisation different semantics;
- any additional option has to be implemented in all the affected INSPIRE-aware software packages.

The logic for assessing the equivalence of texts is going to be a custom logic in any case.
The trimming is implemented in almost every programming language, while the removal of whitespace between words is not so common and "standardised".
There are additional difficulties that come with UTF-8 where some weirdly encoded whitespace characters exist.

Instead, the use of standard xml fragments would actually allow both a strict validation (via the HTTP URI in the xlink:href) while keeping the door open to  humanly acceptable variations in the text.

**Comment by IGNCNIG** [comment]
In our opinion, the best choice is to be strict and require the text as it is in the enumeration, allowing only for different capitalisation.
Therefore, It should be modified the enumeration of Spain (http://inspire.ec.europa.eu/schemas/common/1.0/enums/enum_spa.xsd) with the right text:  "REGLAMENTO (UE) No 1089/2010 DE LA COMISIÓN de 23 de noviembre de 2010 por el que se aplica la Directiva 2007/2/CE del Parlamento Europeo y del Consejo en lo que se refiere a la interoperabilidad de los conjuntos y los servicios de datos espaciales"

**Comment by thijsbrentjens** [comment]
Agree with @IGNCNIG

**Comment by aquaglia** [comment]
I certainly agree on being strict but I do not see the exception for capitalisation as opposed to whitespace.
Please read my comment above.
In the German language, for example, you shall capitalise all nouns.
I do not even know what rules might apply for the Bulgarian, Estonian, Latvian languages, among others.

When, long ago, I decided to include the labels in the XML schema for strict validation it was because I thought that it was more important to be consistent than to use the perfect label especially because there was no source of truth at the time other than copy and pasting from Regulations translations PDFs which sometimes contained weird character encodings.

I think we should agree on a text and that's it, verbatim.

However, the ultimate solution is to use a URI, and linking citation fragments would allow that.

**Comment by AntoRot** [comment]
I agree too with @IGNCNIG on the alternative approach proposed, but it could investigate for once for what languages the capitalisation has a semantic value (as highligthed by @aquaglia) in order to be strict also for capitalisation only for that languages.
I also think that the better ultimate solution is to use the URI, as proposed by @aquaglia, e.g. coming from the ELI (European Legislation Identifier) system, as also recommended in the TG v. 2.0 (see TG rec 5.5).

**Comment by IGNCNIG** [comment]
Do you know when is it going to be fixed? Spanish metadata is having this error yet.

**Comment by LauraAlemany** [comment]
Hello!

Do you know when is this implementation going to be approved? Do you need something we can do? All Spanish metadata is having this error when validating.

Thank you.

Regards,
Laura

**Comment by inakidiazdecerio** [comment]
Hi @LauraAlemany,

All changes are approved and are under testing in the staging environment:

http://etf-staging.eu-west-1.elasticbeanstalk.com/etf-webapp/

We will appreciate your validation over this issue. We will come back to you when the solution is implemented in the production environment.

Regards.

**Comment by LauraAlemany** [comment]
Great!

I have passed one of our metadata and there is no error in the validation.

http://etf-staging.eu-west-1.elasticbeanstalk.com/etf-webapp/v2/TestRuns/EIDa0324a74-9788-412b-b220-4b1a374ae5a3.html 

Thank you :)

Regards. 
Laura

**Comment by kkoistinen** [comment]
I have the same issue with service http://www.paikkatietohakemisto.fi/geonetwork/srv/en/csw?service=CSW&request=GetCapabilities. Error message from staging environment:

Schema not valid: [org.xml.sax.SAXException: Fatal error: org.xml.sax.SAXParseException; lineNumber: 274; columnNumber: 91; cvc-enumeration-valid: Value 'COMMISSION REGULATION (EU) No 1089/2010 of 23 November 2010
            implementing Directive 2007/2/EC of the European Parliament and of the Council as
            regards interoperability of spatial data sets and services' is not facet-valid with respect to enumeration '[COMMISSION REGULATION (EU) No 1089/2010 of 23 November 2010 implementing Directive 2007/2/EC of the European Parliament and of the Council as regards interoperability of spatial data sets and services]'. It must be a value from the enumeration. Response did not validate against schema 'http://www.opengis.net/cat/csw/2.0.2 http://schemas.opengis.net/csw/2.0.2/CSW-discovery.xsd http://inspire.ec.europa.eu/schemas/inspire_ds/1.0 http://inspire.ec.europa.eu/schemas/inspire_ds/1.0/inspire_ds.xsd'.]

**Comment by josemasensio** [comment]
Dear @kkoistinen,

Thank you for your message. Due to the changes that we had made into the validator, we will appreciate if you can open a new issue. 
If it is your first issue, we will be thankful if you can follow our [contribution guidelines](https://github.com/inspire-eu-validation/community/blob/master/contribution_guidelines.md) to help us checking and verifying what happend. 
Thanks in advance.

Best regards.
